src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -173,70 +176,60 @@ private bool CanUseVar(ITypeSymbol typeSymbol)
         {
             var oldBody = arrowExpression;
             var oldParentingNode = oldBody.Parent;
             var leadingTrivia = oldBody.GetLeadingTrivia()
                                        .AddRange(oldBody.ArrowToken.TrailingTrivia);
 
-            var newStatement = Rewrite(document, expression, newLocalName, document, oldBody.Expression, allOccurrences, cancellationToken);
-            var newBody = SyntaxFactory.Block(declarationStatement, SyntaxFactory.ReturnStatement(newStatement))
+            var newExpression = Rewrite(document, expression, newLocalName, document, oldBody.Expression, allOccurrences, cancellationToken);
+
+            var convertedStatement = createReturnStatement
+                ? SyntaxFactory.ReturnStatement(newExpression)
+                : SyntaxFactory.ExpressionStatement(newExpression) as StatementSyntax;
+
+            var newBody = SyntaxFactory.Block(declarationStatement, convertedStatement)
                                        .WithLeadingTrivia(leadingTrivia)
                                        .WithTrailingTrivia(oldBody.GetTrailingTrivia())
                                        .WithAdditionalAnnotations(Formatter.Annotation);
 
-            SyntaxNode newParentingNode = null;
-            if (oldParentingNode is BasePropertyDeclarationSyntax baseProperty)
+            SyntaxNode newParentingNode = oldParentingNode; // in case we fail for some reason, at least don't remove it!
+            
+            switch (oldParentingNode)
             {
-                var getAccessor = SyntaxFactory.AccessorDeclaration(SyntaxKind.GetAccessorDeclaration, newBody);
-                var accessorList = SyntaxFactory.AccessorList(SyntaxFactory.List(new[] { getAccessor }));
+                case BasePropertyDeclarationSyntax baseProperty:
+                    var getAccessor = SyntaxFactory.AccessorDeclaration(SyntaxKind.GetAccessorDeclaration, newBody);
+                    var accessorList = SyntaxFactory.AccessorList(SyntaxFactory.List(new[] { getAccessor }));
 
-                newParentingNode = baseProperty.RemoveNode(oldBody, SyntaxRemoveOptions.KeepNoTrivia);
-
-                if (newParentingNode.IsKind(SyntaxKind.PropertyDeclaration))
-                {
-                    var propertyDeclaration = ((PropertyDeclarationSyntax)newParentingNode);
-                    newParentingNode = propertyDeclaration
+                    newParentingNode = baseProperty
+                        .RemoveNode(oldBody, SyntaxRemoveOptions.KeepNoTrivia)
                         .WithAccessorList(accessorList)
+                        .TryWithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
+                        .WithTrailingTrivia(baseProperty.TryGetSemicolonToken().TrailingTrivia);
+                    break;
+                case AccessorDeclarationSyntax accessor:
+                    newParentingNode = accessor
+                        .RemoveNode(oldBody, SyntaxRemoveOptions.KeepNoTrivia)
```

Comment:
the removenode parts seems weird.  I'm not sure what you're trying to accomplish here.  I feel like we'd want the concept to be "move all the leading/trailing trivia from the original accessor over to the new accessor.  Is that right?

Reply:
it does... but i'm not the one who wrote it... as i said, if i could i would probably rewrite the whole thing

Reply:
> it does... but i'm not the one who wrote it... as i said, if i could i would probably rewrite the whole thing

You are allowed to do this :)